TombEngine/Specific/Worker.cpp

```diff
@@ -0,0 +1,144 @@
+#include "framework.h"
+#include "Specific/Worker.h"
+
+namespace TEN::Utils
+{
+	WorkerManager g_Worker = WorkerManager();
+
+	WorkerManager::WorkerManager()
+	{
+		// Reserve threads.
+		unsigned int threadCount = std::thread::hardware_concurrency() * 2;
+		_threads.reserve(threadCount);
+
+		// Create threads.
+		for (int i = 0; i < threadCount; i++)
+			_threads.push_back(std::thread(&WorkerManager::Worker, this));
+
+		_deinitialize = false;
+	}
+
+	WorkerManager::~WorkerManager()
+	{
+		// LOCK: Restrict shutdown flag access.
+		{
+			auto taskLock = std::lock_guard(_taskMutex);
+
+			_deinitialize = true;
+		}
+
+		// Notify all threads they should stop.
+		_taskCond.notify_all();
+
+		// Join all threads.
+		for (auto& thread : _threads)
+		{
+			if (thread.joinable())
+				thread.join();
```

Comment:
Be careful, thread::join can throw, and destructors are not allowed to throw exceptions (it is UB to do so)